### PR TITLE
RenderObject.dropChild() should verify child was removed already

### LINKED ISSUE #1256: RenderObject.dropChild() should verify child was removed already
dropChild() is an internal API, but currently it's easy to call it directly.
We should have it assert that visitChildren() doesn't visit the child.

Similarly with adoptChild, in reverse.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #1256

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/rendering/object.dart [hunk 1]
```diff
@@ -687,6 +687,15 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
       child.parentData = new ParentData();
   }
 
+  bool _debugHasChild(RenderObject child) {
+    bool foundChild = false;
+    visitChildren((RenderObject otherChild) {
+      if (otherChild == child)
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: assert found child is false

**Comment by Hixie** [comment]
can you check that abstract node checks the parent pointer is null?

are there cases the parent pointer is null and yet the child still gets walked?

maybe the visit children api should assert the children have parent=this

**Comment by Hixie** [comment]
lgtm but Travis is mad

**Comment by abarth** [comment]
Apparently this assert fires all the time.  /me will study more.

**Pull request closed without merging**